pennylane/drawer/draw.py

```diff
@@ -178,19 +180,42 @@ def transformed_circuit(x):
 
         0: ──RX(0.90)─┤  <Z>
 
+    The function also accepts quantum functions rather than QNodes. This can be especially
+    helpful if you want to visualize only a part of a circuit that may not be convertible into
+    a QNode, such as a sub-function that does not return any measurements.
+
+    >>> def qfunc(x):
+    ...     qml.RX(x, wires=[0])
+    ...     qml.CNOT(wires=[0, 1])
+    >>> print(qml.draw(qfunc)(1.1))
+    0: ──RX(1.10)─╭●─┤
+    1: ───────────╰X─┤
+
     """
 
     @wraps(qnode)
     def wrapper(*args, **kwargs):
-        original_expansion_strategy = getattr(qnode, "expansion_strategy", None)
-
-        try:
-            qnode.expansion_strategy = expansion_strategy or original_expansion_strategy
-            tapes = qnode.construct(args, kwargs)
-        finally:
-            qnode.expansion_strategy = original_expansion_strategy
-
-        _wire_order = wire_order or qnode.device.wires
+        if hasattr(qnode, "construct"):

```

Comment:
did this instead of `isinstance(qnode, QNode)` because batch transforms create non-qnodes that have a `construct` method. Of course, those don't have a `qtape` attribute, so I have to do an `if not tapes` check below. I also use the same `hasattr` check in `draw_mpl`, though I'm not sure it's necessary because it doesn't appear to support batch-transformed qnodes. That said, I'd rather leave it that way for consistency.

Reply:
@timmysilv this was originally a 'hack' by me ages ago because lots of people were requesting the ability to `draw(some_batch_transform(qnode))`. 

I had it on my list to improve, but it requires a change to batch transforms. Namely, that `batch_transforms` always _returns_ a QNode (which in turn requires formalizing that QNodes can contain multiple scripts). 

So lots of moving parts.

Reply:
good context, ty! wondering if you have feelings for what should happen in this PR? I think I would like to leave it as is, but if we have an improvement plan for batch transforms, this code should certainly be considered

Reply:
Yep 100%! I'll tag @rmoyard here who is thinking about this

Reply:
Thank you for pointing this, indeed it is not ideal behaviour!